src/test/java/com/puppycrawl/tools/checkstyle/xpath/ElementNodeTest.java

```diff
@@ -58,6 +58,14 @@ public void init() throws Exception {
         rootNode = new RootNode(rootAst);
     }
 
+    @Test
+    public void testCompareOrderWrongInstance() throws Exception {
+        final String xpath = "//OBJBLOCK";
+        final List<NodeInfo> nodes = getXpathItems(xpath, rootNode);
+        final int result = nodes.get(0).compareOrder(null);
```

Comment:
Why is there a check for `null`? Does this not happen naturally?

Reply:
No, naturally such a case doesn't arise, so this was added to achieve 100% branch coverage.

Reply:
If `null` isn't natural and doc doesn't say it can send a null then why do we have a check for it. I assume the branch was the null check? If it can never be null then the check is pointless and can be removed.

Reply:
the `if (other instanceof ElementNode)` is required by Spotbugs otherwise it will throw an `unchecked cast` error for `((ElementNode) other).indexAmongSiblings`. So, I have replaced `null` with `rootNode` object instead to fail it and cover the branch.

Reply:
Done